### PR TITLE
Consume TAP in Avocado with a specialized loader

### PR DESCRIPTION
**Pull request by bonzini**
This pull request adds a new loader that parses TAP output from a file, logs individual test results and converts the result into a global PASS/FAIL/SKIP outcome.

The first part of the pull request is actually a refactoring of FileLoader to simplify the creation of alternative loaders. The newly introduced class, SimpleFileLoader, is then used by the TAP plugin, but it could also be used by e.g. the GLib loader in order to get better error reporting (e.g. broken links) and remove duplicated code.

### CODE REVIEW

**Comment by bonzini** [comment]
The Travis failure seems preexisting.

File: optional_plugins/tap/avocado_tap/__init__.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                                   [Runnable('tap', reference)])
+
+
+class TapCLI(CLI):
+
+    """
+    Run Test Anything Protocol tests
+    """
+
+    name = 'tap'
```
Comment (beraldoleal) [inline optional_plugins/tap/avocado_tap/__init__.py]: @clebergnu or @bonzini, we already have a `name = 'TAP'` from CLI on `plugins/tap.py`.

Will this name conflict with the current plugin or not?

Comment (clebergnu) [reply optional_plugins/tap/avocado_tap/__init__.py]: This name is purely informational, so it won't cause a hard conflict other than possibly confusing users about two plugins of the same type and same name.  The confusion is highly minimized by the different descriptions, though.

Comment (beraldoleal) [reply optional_plugins/tap/avocado_tap/__init__.py]: Yes, I was talking about a "soft conflict", two plugins with the same name.

Comment (clebergnu) [reply optional_plugins/tap/avocado_tap/__init__.py]: Yes, no conflict is better than a soft conflict indeed.  If possible, we should rename those, such as `tap_results` and `tap_loader` IMO.

**Comment by clebergnu** [comment]
Superseded by https://github.com/avocado-framework/avocado/pull/3417

**Pull request closed without merging**